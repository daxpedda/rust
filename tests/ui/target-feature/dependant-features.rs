//@ build-fail
//@ only-wasm32-wasip1
//@ compile-flags: --crate-type=lib

pub fn test() {
    #[target_feature(enable = "relaxed-simd")]
    //~^ ERROR the target feature relaxed-simd requires simd128 to be enabled
    unsafe fn inner() {}

    unsafe {
        foo();
        bar();
        baz();
        inner();
    }
}

#[target_feature(enable = "relaxed-simd")]
//~^ ERROR the target feature relaxed-simd requires simd128 to be enabled
unsafe fn foo() {}

#[target_feature(enable = "relaxed-simd,simd128")]
unsafe fn bar() {}

#[target_feature(enable = "relaxed-simd")]
#[target_feature(enable = "simd128")]
unsafe fn baz() {}
