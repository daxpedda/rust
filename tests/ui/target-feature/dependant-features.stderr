error: the target feature relaxed-simd requires simd128 to be enabled
  --> $DIR/tied-features.rs:6:5
   |
LL |     #[target_feature(enable = "relaxed-simd")]
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: add the missing features in a `target_feature` attribute

error: the target feature relaxed-simd requires simd128 to be enabled
  --> $DIR/tied-features.rs:18:1
   |
LL | #[target_feature(enable = "relaxed-simd")]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: add the missing features in a `target_feature` attribute

error: aborting due to 2 previous errors

